backend/main/src/Shared/Adapters/DB/NishikiTableClient.ts

```diff
@@ -104,6 +106,124 @@ export class NishikiDynamoDBClient {
 		await this.dynamoClient.send(command);
 	}
 
+	/**
+	 * Get a group from the DB.
+	 * @param groupId
+	 * @returns {GroupData | null} - the group data. If the group does not exist, it returns null.
+	 */
+	async getGroup(groupId: string): Promise<GroupData | null> {
+		const getGroupInput: GetItemInput = {
+			TableName: this.tableName,
+			Key: marshall({
+				PK: groupId,
+				SK: "Group",
+			}),
+		};
+
+		const command = new GetItemCommand(getGroupInput);
+		const response = await this.dynamoClient.send(command);
+
+		if (!response.Item) return null;
+
+		const unmarshalledData = unmarshall(response.Item);
+
+		return {
+			groupId: unmarshalledData.PK,
+			groupName: unmarshalledData.GroupName,
+		};
+	}
+
+	/**
+	 * Save a group to the DB.
+	 * This function generates multiple putItem commands and sends it concurrently.
+	 * If user IDs are provided, this function issues [PutItem commands for creating user and group relations](https://genesis-tech-tribe.github.io/nishiki-documents/project-document/database#user).
+	 * @param groupId
+	 * @param props
+	 */
+	async saveGroup(groupId: string, props: GroupInput) {
+		const { groupName, userIds, containers } = props;
```

Comment:
This `containers` include only container ID, so its name should be containerIds like userIds.